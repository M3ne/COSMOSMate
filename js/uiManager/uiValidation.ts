import $ from 'jquery';

export const allowPeriod = (e: KeyboardEvent) => {
  if (e.code === 'Period' || e.code === 'NumpadDecimal') e.preventDefault();
};
export const esDay366 = () => {
  if ($('#es-day').val() < 0) $('#es-day').val('000.00000000');
  if ($('#es-day').val() >= 367) $('#es-day').val('366.00000000');
};
export const esInc180 = function (): void {
  if ($('#es-inc').val() < 0) $('#es-inc').val('000.0000');
  if ($('#es-inc').val() > 180) $('#es-inc').val('180.0000');
};
export const esRasc360 = function (): void {
  if ($('#es-rasc').val() < 0) $('#es-rasc').val('000.0000');
  if ($('#es-rasc').val() > 360) $('#es-rasc').val('360.0000');
};
export const esMeanmo18 = function (): void {
  if ($('#es-meanmo').val() < 0) $('#es-meanmo').val('00.00000000');
  if ($('#es-meanmo').val() > 18) $('#es-meanmo').val('18.00000000');
};
export const esArgPe360 = function (): void {
  if ($('#es-argPe').val() < 0) $('#es-argPe').val('000.0000');
  if ($('#es-argPe').val() > 360) $('#es-argPe').val('360.0000');
};
export const esMeana360 = function (): void {
  if ($('#es-meana').val() < 0) $('#es-meana').val('000.0000');
  if ($('#es-meana').val() > 360) $('#es-meana').val('360.0000');
};
export const msLat90 = function (): void {
  if ($('#ms-lat').val() < -90) $('#ms-lat').val('-90.000');
  if ($('#ms-lat').val() > 90) $('#ms-lat').val('90.000');
};
export const msLon180 = function (): void {
  if ($('#ms-lon').val() < -180) $('#ms-lon').val('-180.000');
  if ($('#ms-lon').val() > 180) $('#ms-lon').val('180.000');
};
export const initUiValidation = () => {
  $('#editSat>div>input').on('keydown', validateNumOnly);
  $('#es-ecen').on('keydown', allowPeriod);
  $('#es-day').on('keyup', esDay366);
  $('#es-inc').on('keyup', esInc180);
  $('#es-rasc').on('keyup', esRasc360);
  $('#es-meanmo').on('keyup', esMeanmo18);
  $('#es-argPe').on('keyup', esArgPe360);
  $('#es-meana').on('keyup', esMeana360);
  $('#ms-lat').on('keyup', msLat90);
  $('#ms-lon').on('keyup', msLon180);
};
export const validateNumOnly = (e: KeyboardEvent) => {
  if (
    // Allow: backspace, delete, tab, escape, enter and .
    $.inArray(e.code, ['Delete', 'Backspace', 'Tab', 'Escape', 'NumpadEnter', 'Enter', 'NumpadEnter', 'Period']) !== -1 ||
    // Allow: Ctrl+A, Command+A
    (e.code === 'KeyA' && (e.ctrlKey === true || e.metaKey === true)) ||
    // Allow: home, end, left, right, down, up
    $.inArray(e.code, ['Home', 'End', 'ArrowLeft', 'ArrowUp', 'ArrowRight', 'ArrowDown']) !== -1
  ) {
    // let it happen, don't do anything
    return;
  }
  // Ensure that it is a number and stop the keypress
  if (
    $.inArray(e.code, ['Digit1', 'Digit2', 'Digit3', 'Digit4', 'Digit5', 'Digit6', 'Digit7', 'Digit8', 'Digit9', 'Digit0']) !== -1 ||
    $.inArray(e.code, ['Numpad1', 'Numpad2', 'Numpad3', 'Numpad4', 'Numpad5', 'Numpad6', 'Numpad7', 'Numpad8', 'Numpad9', 'Numpad0']) !== -1
  ) {
    e.preventDefault();
  }
};
