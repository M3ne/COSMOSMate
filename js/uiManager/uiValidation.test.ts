import * as uiValidation from './uiValidation';
describe('uiValidation.validateNumOnly', () => {
  test('0', () => {
    let result = uiValidation.validateNumOnly(<KeyboardEvent>{ code: 'Numpad0', preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
  test('1', () => {
    let result = uiValidation.validateNumOnly(<KeyboardEvent>{ code: 'Digit5', preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
  test('2', () => {
    let result = uiValidation.validateNumOnly(<KeyboardEvent>{ code: 'Home', preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
  test('3', () => {
    let result = uiValidation.validateNumOnly(<KeyboardEvent>{ code: 'Backspace', preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
  test('4', () => {
    let result = uiValidation.validateNumOnly(<KeyboardEvent>{ code: 'Enter', preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
  test('5', () => {
    let result = uiValidation.validateNumOnly(<KeyboardEvent>{ code: 'KeyA', ctrlKey: true, preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
  test('6', () => {
    let result = uiValidation.validateNumOnly(<KeyboardEvent>{ code: 'KeyA', ctrlKey: false, preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
});

describe('uiValidation.allowPeriod', () => {
  test('0', () => {
    let result = uiValidation.allowPeriod(<KeyboardEvent>{ code: 'KeyA', ctrlKey: false, preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result = uiValidation.allowPeriod(<KeyboardEvent>{ code: 'Period', preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result = uiValidation.allowPeriod(<KeyboardEvent>{ code: 'NumpadDecimal', preventDefault: () => {} });
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.esDay366', () => {
  test('0', () => {
    let result: any = uiValidation.esDay366();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    document.body.innerHTML = `<input id="es-day" value="367" />`;
    let result: any = uiValidation.esDay366();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="es-day" value="-1" />`;
    let result: any = uiValidation.esDay366();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.esInc180', () => {
  test('0', () => {
    let result: any = uiValidation.esInc180();
    expect(result).toMatchSnapshot();
  });
  test('1', () => {
    document.body.innerHTML = `<input id="es-inc" value="181" />`;
    let result: any = uiValidation.esInc180();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="es-inc" value="-1" />`;
    let result: any = uiValidation.esInc180();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.esRasc360', () => {
  test('0', () => {
    let result: any = uiValidation.esRasc360();
    expect(result).toMatchSnapshot();
  });
  test('1', () => {
    document.body.innerHTML = `<input id="es-rasc" value="361" />`;
    let result: any = uiValidation.esRasc360();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="es-rasc" value="-1" />`;
    let result: any = uiValidation.esRasc360();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.esMeanmo18', () => {
  test('0', () => {
    let result: any = uiValidation.esMeanmo18();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    document.body.innerHTML = `<input id="es-meanmo" value="361" />`;
    let result: any = uiValidation.esMeanmo18();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="es-meanmo" value="-1" />`;
    let result: any = uiValidation.esMeanmo18();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.esArgPe360', () => {
  test('0', () => {
    let result: any = uiValidation.esArgPe360();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    document.body.innerHTML = `<input id="es-argPe" value="361" />`;
    let result: any = uiValidation.esArgPe360();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="es-argPe" value="-1" />`;
    let result: any = uiValidation.esArgPe360();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.esMeana360', () => {
  test('0', () => {
    let result: any = uiValidation.esMeana360();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    document.body.innerHTML = `<input id="es-meana" value="361" />`;
    let result: any = uiValidation.esMeana360();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="es-meana" value="-1" />`;
    let result: any = uiValidation.esMeana360();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.msLat90', () => {
  test('0', () => {
    let result: any = uiValidation.msLat90();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    document.body.innerHTML = `<input id="ms-lat" value="-91" />`;
    let result: any = uiValidation.msLat90();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="ms-lat" value="91" />`;
    let result: any = uiValidation.msLat90();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.msLon180', () => {
  test('0', () => {
    let result: any = uiValidation.msLon180();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    document.body.innerHTML = `<input id="ms-lon" value="181" />`;
    let result: any = uiValidation.msLon180();
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    document.body.innerHTML = `<input id="ms-lon" value="-181" />`;
    let result: any = uiValidation.msLon180();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('uiValidation.initUiValidation', () => {
  test('0', () => {
    let result: any = uiValidation.initUiValidation();
    expect(result).toMatchSnapshot();
  });
});
