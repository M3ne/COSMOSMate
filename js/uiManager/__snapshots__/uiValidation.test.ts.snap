// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiValidation.allowPeriod 0 1`] = `undefined`;

exports[`uiValidation.allowPeriod 1 1`] = `undefined`;

exports[`uiValidation.allowPeriod 2 1`] = `undefined`;

exports[`uiValidation.esArgPe360 0 1`] = `undefined`;

exports[`uiValidation.esArgPe360 1 1`] = `undefined`;

exports[`uiValidation.esArgPe360 2 1`] = `undefined`;

exports[`uiValidation.esDay366 0 1`] = `undefined`;

exports[`uiValidation.esDay366 1 1`] = `undefined`;

exports[`uiValidation.esDay366 2 1`] = `undefined`;

exports[`uiValidation.esInc180 0 1`] = `undefined`;

exports[`uiValidation.esInc180 1 1`] = `undefined`;

exports[`uiValidation.esInc180 2 1`] = `undefined`;

exports[`uiValidation.esMeana360 0 1`] = `undefined`;

exports[`uiValidation.esMeana360 1 1`] = `undefined`;

exports[`uiValidation.esMeana360 2 1`] = `undefined`;

exports[`uiValidation.esMeanmo18 0 1`] = `undefined`;

exports[`uiValidation.esMeanmo18 1 1`] = `undefined`;

exports[`uiValidation.esMeanmo18 2 1`] = `undefined`;

exports[`uiValidation.esRasc360 0 1`] = `undefined`;

exports[`uiValidation.esRasc360 1 1`] = `undefined`;

exports[`uiValidation.esRasc360 2 1`] = `undefined`;

exports[`uiValidation.initUiValidation 0 1`] = `undefined`;

exports[`uiValidation.msLat90 0 1`] = `undefined`;

exports[`uiValidation.msLat90 1 1`] = `undefined`;

exports[`uiValidation.msLat90 2 1`] = `undefined`;

exports[`uiValidation.msLon180 0 1`] = `undefined`;

exports[`uiValidation.msLon180 1 1`] = `undefined`;

exports[`uiValidation.msLon180 2 1`] = `undefined`;

exports[`uiValidation.validateNumOnly 0 1`] = `undefined`;

exports[`uiValidation.validateNumOnly 1 1`] = `undefined`;

exports[`uiValidation.validateNumOnly 2 1`] = `undefined`;

exports[`uiValidation.validateNumOnly 3 1`] = `undefined`;

exports[`uiValidation.validateNumOnly 4 1`] = `undefined`;

exports[`uiValidation.validateNumOnly 5 1`] = `undefined`;

exports[`uiValidation.validateNumOnly 6 1`] = `undefined`;
